### DISCUSSION TITLE
Dispatch based on path param types

### DISCUSSION THREAD

**Discussion by mhagnumdw** [microcks/microcks#654] [Q&A] [answered]
The routes below conflict:

1. /iptu/debitos/{id}/anos
2. /iptu/debitos/{id}/{year}

`{id}` and `{year}` are numeric. When accessing `/iptu/debitos/123/anos` it is sometimes redirected to route 2 instead of 1.

Is it possible to decide the route based on type or regular expression maybe?

**Comment by jacksoncastro** [comment]
I I had the same problem. Any suggestion?

**Reply by lbroudoux** [reply]
Hi there!

Operation matching is solely based on resource paths extracted from your example. So in theory, as long as you don't have an example having `anos` as the value for `year` it shouldn't conflict...

That said this is theory 😅 Would you share an OpenAPI spec to reproduce this? I'll have a look at it.

**Reply by mhagnumdw** [reply]
Hi @lbroudoux!

OpenAPI spec: [debitos-api.zip](https://github.com/microcks/microcks/files/9466203/debitos-api.zip)

This is ok:
```bash
curl https://......../api/debitos/123/2022 | jq
```
Changing only the last parameter from 2022 to 2023, the return becomes that of the other endpoint.

**Reply by lbroudoux** [reply]
Hi @mhagnumdw 
I suspect you're using version `1.5.2` and it appears that we had a regression on this kind of dispatch. See #619 - title is a bit different but the root cause is similar. I've tested with soon-to-be-released `1.6.0` and it behaves correctly.

Would you be able to test on that one? The tag of the corresponding container image is `nightly` at the moment. Depending on how you install Microcks, you'll have to update your compose file or values.yaml or CRD.

Let me know.

**Reply by lbroudoux** [reply]
Correct behavior is now applied: when a dispatch criterion is not found / unknown it defaults to a random response in the same operation. If you want to override this behavior, you can still use a `FALLBACK` dispatcher.

```sh
$ curl http://localhost:8080/rest/debitos/1.0.0/api/debitos/123/2023 -s | jq .                                                                                                                                                              
{
  "idIptuImovel": 1234,
  "iptuLiberado": true,
  "iptuImovelIsento": true,
  "iptuImovelSuspento": true,
  "iptuImovelExtinto": true,
  "existeRescisaoDebito": true,
  "indicatorMensagemSimples": true,
  "debitoEnviadoPGM": true,
  "permiteExibirDam": true,
  "iptuImovelMensagem": "Mensagem aqui",
  "numeroDocumento": "042.123.232-80",
  "mensagemCotaUnica": "Para liquidar...",
  "mensagemRegularizarDebito": "Mensagem",
  "dataLimitePagamento": "2020-08-30",
  "mensagemRegraIncentivo": "Mensagem",
  "mensagemRegraIncentivoInformativa": "Mensagem",
  "exibirParcelado": true,
  "parcelas": [
    {
      "vencimento": 2022,
      "valorParcela": 0,
      "valorCorrecao": 0,
      "valorMulta": 0,
      "valorJuros": 0,
      "valorAPagar": 0,
      "receberAte": "2020-08-30",
      "situacao": "Em atraso"
    }
  ]
}
```

**Comment by mhagnumdw** [comment, marked as answer]
> Would you be able to test on that one? The tag of the corresponding container image is nightly at the moment. Depending on how you install Microcks, you'll have to update your compose file or values.yaml or CRD.

Hi @lbroudoux , version 1.6.0-SNAPSHOT (quay.io/microcks/microcks-ansible-operator:nightly) worked perfectly. **Thank you very much!!** :partying_face:  :partying_face:  :partying_face: 

ps: I had to edit the CRD manually in Openshift as there is no option to choose another version. I also tried installing another version of operator, but the only option is stable.

![image](https://user-images.githubusercontent.com/17688846/187917401-15d767a1-2965-489d-8660-9d67a2f4c16d.png)
